File: src/java/org/javaturk/dp/ch02/factory/solution/Employee.java
package org.javaturk.dp.ch02.factory.solution;

public class Employee implements Worker {
	private static final int BASE_SALARY = 500;
	protected int no;
	protected String name;
	protected int year;
	protected String department;

	public Employee(int no, String name, int year, String department) {
		this.no = no;
		this.name = name;
		this.year = year;
		this.department = department;
	}

	public Employee(int no, String name, int year) {
		this.no = no;
		this.name = name;
		this.year = year;
	}
	
	@Override
	public void work() {
		System.out.println("Employee is working!");
	}

	public void printInfo() {
		System.out.println("\nNo: " + no);
		System.out.println("Name:" + name);
		System.out.println("Year: " + year);
		System.out.println("Department: " + department);
	}

//	public int getNo() {
//		return no;
//	}
//
//	public void setNo(int no) {
//		this.no = no;
//	}
//
//	public String getName() {
//		return name;
//	}
//
//	public void setName(String name) {
//		this.name = name;
//	}
//
//	public int getYear() {
//		return year;
//	}
//
//	public void setYear(int year) {
//		this.year = year;
//	}
//
//	public String getDepartment() {
//		return department;
//	}
//
//	public void setDepartment(String department) {
//		this.department = department;
//	}

	@Override
	public String toString() {
		return "Employee [no=" + no + ", name=" + name + ", year=" + year + ", department=" + department + "]";
	}

	public double calculateSalary() {
		return year * BASE_SALARY;
	}
}

File: src/c#/Creational/FactoryMethod/Bloch/Employee.cs
using System;
using System.Text;

namespace Bloch
{
    public class Employee
    {
        public int no { get; set; }
        public String name { get; set; }
        public int year { get; set; }
        public string department { get; set; }
        public double salary { get; set; }

        public static readonly int BASE_SALARY = 500;

        private Employee(int no, String name, int year, string department)
        {
            this.no = no;
            this.name = name;
            this.year = year;
            this.department = department;
        }

        public static Employee CreateNewEmployee(int no, String name, String department)
        {
            return new Employee(no, name, 0, department);
        }

        public static Employee CreateNewTemporaryEmployee(int no, String name)
        {
            return new Employee(no, name, 0, "Temp");
        }

        public virtual void Work()
        {
            Console.WriteLine("Employee is working!");
        }

        public virtual double CalculateSalary()
        {
            salary = year * BASE_SALARY;
            return salary;
        }

        public override string ToString()
        {
            StringBuilder builder = new StringBuilder("\nEmployee Info");
            builder.Append("No: " + no);
            builder.Append("Name:" + name);
            builder.Append("Year: " + year);
            builder.Append("Department: " + department);
            builder.Append("Salary: " + CalculateSalary());
            return builder.ToString();
        }
    }
}
